### PR TITLE
Fix sky spot launch TPU

### PR DESCRIPTION
**Pull request by infwinston**
Now `sky spot launch` a TPU VM will fail due to `gcloud` not fully setup on the spot controller.
This PR fixes it by uploading `~/.config/gcloud/active_config`.

### CODE REVIEW

**Comment by infwinston** [comment]
Just added!

File: tests/test_smoke.py [hunk 1]
```diff
@@ -705,6 +705,24 @@ def test_spot_storage():
...
+    test = Test(
+        'test-spot-tpu',
+        [
+            f'sky spot launch -n {name} examples/tpu/tpuvm_mnist.yaml -y -d',
+            'sleep 5',
+            f's=$(sky spot status); printf "$s"; echo; echo; printf "$s" | grep {name} | head -n1 | grep STARTING',
+            'sleep 600',  # TPU takes a while to launch
+            f's=$(sky spot status); printf "$s"; echo; echo; printf "$s" | grep {name} | head -n1 | grep "RUNNING\|SUCCEEDED"',
+        ],
+        f'sky spot cancel -y -n {name}',
```
Comment (concretevitamin) [inline tests/test_smoke.py:721]: We don't need a tear down command for managed spot jobs.

**Review by concretevitamin — APPROVED** [review]
LGTM - with this, does the user example run successfully as a managed spot job?

File: tests/test_smoke.py (hunk 1 above)
Comment (infwinston) [reply tests/test_smoke.py:721]: Oh this smoke test will pass if the job changed from `STARTING` to `RUNNING`. In this case, we can actually kill the job to save some costs. But we can remove it if we want as the job is short.

**Comment by infwinston** [comment]
> LGTM - with this, does the user example run successfully as a managed spot job?

Yes, it does run successfully now.

File: tests/test_smoke.py (hunk 1 above)
Comment (concretevitamin) [reply tests/test_smoke.py:721]: Oh I thought it was short - let's still keep it then.

**Comment by Michaelvll** [comment]
This may change a bit of the authentication of the GCP account. Please run smoke tests before merging the PR. ; )

**Comment by infwinston** [comment]
On RISE server, smoke test passed except for the below one. Offline discussion with @Michaelvll suggests this is not related to the PR and I file one issue at https://github.com/skypilot-org/skypilot/issues/1266. I'm merging this change now.

**Pull request merged**